cdc/owner/barrier.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package owner
+
+import (
+	"math"
+
+	"github.com/pingcap/log"
+	"github.com/pingcap/ticdc/cdc/model"
+)
+
+type barrierType = int
+
+const (
+	// DDLJobBarrier denotes a replication barrier caused by a DDL.
+	DDLJobBarrier barrierType = iota
+	// SyncPointBarrier denotes a barrier for snapshot replication.
+	SyncPointBarrier
+	// FinishBarrier denotes a barrier for changefeed finished.
+	FinishBarrier
+)
+
+type barriers struct {
```

Comment:
Could you add some comments about the barrier and its thread safety? Because I'm confused at the first glance, in most cases, a barrier is thread-safe but in this case, it's not.